src/Admin/Pool.php

```diff
@@ -235,20 +235,21 @@ public function hasAdminByClass($class)
      *
      * @throws \InvalidArgumentException if the root admin code is an empty string
      *
-     * @return \Sonata\AdminBundle\Admin\AdminInterface|false
+     * @return AdminInterface|false
      */
     public function getAdminByAdminCode($adminCode)
     {
         if (!\is_string($adminCode)) {
             @trigger_error(sprintf(
-                'Passing a non string value as argument 1 for %s() is deprecated since sonata-project/admin-bundle 3.x and will throw an exception in 4.0.',
+                'Passing a non string value as argument 1 for %s() is deprecated since sonata-project/admin-bundle 3.x and will result in a PHP TypeError in 4.0.',
```

Comment:
I'm :-1: on this. Other throwables we're using are already provided natively by PHP or by SPL. Even, you could throw `\TypeError` from userland code. IMO, we should not make such distinction.

Reply:
I agree, but that's not what I meant here. This is just a textual change here. If in 4.0 `$adminCode` would be typehinted with `string`, then PHP itself will send out a `\TypeError`. There's no intent for 4.0 to throw a `\TypeError` manually here (would not even make sense as PHP will already trigger it and therefor will never hit the code where you would do it manually).